src/librustc/hir/lowering.rs

```diff
@@ -2172,7 +2172,7 @@ impl<'a> LoweringContext<'a> {
                 //             ::std::option::Option::Some(val) => val,
                 //             ::std::option::Option::None => break
                 //           };
-                //           <body>
+                //           <body>;
```

Comment:
Certain expressions (loops? not sure. maybe `if` too) don't require `;` after them.